### PR TITLE
static/scriptengine: use Handle instead of Persistence

### LINKED ISSUE #569: desurium does not build with newest v8
```
[ 18%] Building CXX object src/static/scriptengine/CMakeFiles/scriptengine.dir/code/ScriptCoreInternal.cpp.o
cd /var/tmp/portage/games-util/desurium-9999/work/desurium-9999_build/src/static/scriptengine && /usr/bin/x86_64-pc-linux-gnu-g++  -DBOOST_FILESYSTEM_VERSION=3 -DBOOST_NO_CXX11_SCOPED_ENUMS=1 -DBOOST_NO_SCOPED_ENUMS=1 -DNDEBUG -D_FILE_OFFSET_BITS=64 -D_LARGE_FILES -D__WXGTK__ -DwxDEBUG_LEVEL=0  -DNDEBUG -O2 -pipe -march=native -ftree-vectorize -mavx -maes   -fPIC -pipe -fvisibility=hidden -finline-functions  -fpermissive -std=c++0x -I/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/include -I/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/common -I/var/tmp/portage/games-util/desurium-9999/work/desurium-9999_build/include -I/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine -I/var/tmp/portage/games-util/desurium-9999/work/desurium-9999_build/fetch_cef-prefix/src/fetch_cef    -o CMakeFiles/scriptengine.dir/code/ScriptCoreInternal.cpp.o -c /var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp
/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp: In Elementfunktion »void ScriptCoreInternal::init()«:
/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp:73:85: Warnung: »static v8::Persistent<v8::Context> v8::Context::New(v8::ExtensionConfiguration*, v8::Handle<v8::ObjectTemplate>, v8::Handle<v8::Value>)« ist veraltet (deklariert bei /usr/include/v8.h:4849) [-Wdeprecated-declarations]
In file included from /var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.h:26:0,
                 from /var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp:21:
/usr/include/v8.h:750:3: Fehler: »v8::Persistent<T>::Persistent(const v8::Persistent<T>&) [mit T = v8::Context; v8::Persistent<T> = v8::Persistent<v8::Context>]« ist privat
/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp:73:85: Fehler: in diesem Zusammenhang
In file included from /var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.h:26:0,
                 from /var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp:21:
/usr/include/v8.h:752:3: Fehler: »v8::Persistent<T>& v8::Persistent<T>::operator=(const v8::Persistent<T>&) [mit T = v8::Context; v8::Persistent<T> = v8::Persistent<v8::Context>]« ist privat
/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp:74:16: Fehler: in diesem Zusammenhang
/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp: In Elementfunktion »void ScriptCoreInternal::runString(const char*)«:
/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp:93:46: Fehler: keine passende Funktion für Aufruf von »v8::Context::Scope::Scope(v8::Persistent<v8::Context>&)«
/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp:93:46: Anmerkung: Kandidaten sind:
In file included from /var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.h:26:0,
                 from /var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp:21:
/usr/include/v8.h:4970:5: Anmerkung: v8::Context::Scope::Scope(v8::Isolate*, v8::Persistent<v8::Context>&)
/usr/include/v8.h:4970:5: Anmerkung:   Kandidat erwartet 2 Argumente, 1 angegeben
/usr/include/v8.h:4966:14: Anmerkung: v8::Context::Scope::Scope(v8::Handle<v8::Context>)
/usr/include/v8.h:4966:14: Anmerkung:   keine bekannte Umwandlung für Argument 1 von »v8::Persistent<v8::Context>« nach »v8::Handle<v8::Context>«
/usr/include/v8.h:4964:9: Anmerkung: constexpr v8::Context::Scope::Scope(const v8::Context::Scope&)
/usr/include/v8.h:4964:9: Anmerkung:   keine bekannte Umwandlung für Argument 1 von »v8::Persistent<v8::Context>« nach »const v8::Context::Scope&«
/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp: In Elementfunktion »void ScriptCoreInternal::runScript(const char*, const char*, uint32_t)«:
/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp:122:46: Fehler: keine passende Funktion für Aufruf von »v8::Context::Scope::Scope(v8::Persistent<v8::Context>&)«
/var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp:122:46: Anmerkung: Kandidaten sind:
In file included from /var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.h:26:0,
                 from /var/tmp/portage/games-util/desurium-9999/work/desurium-9999/src/static/scriptengine/code/ScriptCoreInternal.cpp:21:
/usr/include/v8.h:4970:5: Anmerkung: v8::Context::Scope::Scope(v8::Isolate*, v8::Persistent<v8::Context>&)
/usr/include/v8.h:4970:5: Anmerkung:   Kandidat erwartet 2 Argumente, 1 angegeben
/usr/include/v8.h:4966:14: Anmerkung: v8::Context::Scope::Scope(v8::Handle<v8::Context>)
/usr/include/v8.h:4966:14: Anmerkung:   keine bekannte Umwandlung für Argument 1 von »v8::Persistent<v8::Context>« nach »v8::Handle<v8::Context>«
/usr/include/v8.h:4964:9: Anmerkung: constexpr v8::Context::Scope::Scope(const v8::Context::Scope&)
/usr/include/v8.h:4964:9: Anmerkung:   keine bekannte Umwandlung für Argument 1 von »v8::Persistent<v8::Context>« nach »const v8::Context::Scope&«
make[2]: *** [src/static/scriptengine/CMakeFiles/scriptengine.dir/code/ScriptCoreInternal.cpp.o] Fehler 1
make[2]: Leaving directory `/var/tmp/portage/games-util/desurium-9999/work/desurium-9999_build'
make[1]: *** [src/static/scriptengine/CMakeFiles/scriptengine.dir/all] Fehler 2
make[1]: Leaving directory `/var/tmp/portage/games-util/desurium-9999/work/desurium-9999_build'
make: *** [all] Fehler 2
```

### PR DESCRIPTION
**Pull request by karolherbst**
fixes #569

### CODE REVIEW

**Comment by karolherbst** [comment]
this has a compile error with older versions of v8, my bad :(

File: src/static/scriptengine/code/ScriptCoreInternal.cpp [hunk 1]
```diff
@@ -70,7 +70,7 @@ void ScriptCoreInternal::init()
 	global->Set(v8::String::New("Warning"), v8::FunctionTemplate::New(JSWarning));
 	global->Set(v8::String::New("Debug"), v8::FunctionTemplate::New(JSDebug));
 
-	v8::Persistent<v8::Context> context = v8::Context::New(RegisterJSBindings(), global);
+	v8::Handle<v8::Context> context = v8::Context::New(v8::Isolate::GetCurrent(), RegisterJSBindings(), global);
```
Comment (karolherbst) [inline src/static/scriptengine/code/ScriptCoreInternal.cpp]: this is the line which is breaking compatiblity with older versions of v8

**Comment by TheSiege** [comment]
confirmed

**Comment by TheSiege** [comment]
so what do we do about it?

**Comment by karolherbst** [comment]
either I check against the v8 version in source (#ifdef) or I find a way how to convert a v8::Persistent into a v8::Handle

**Comment by karolherbst** [comment]
but I think the first one isn't possible, because they don't provide any #defines for it

**Comment by karolherbst** [comment]
does this error occur without this patch, too?

**Comment by TheSiege** [comment]
i got two white bars in the game symbols apart from that works like a charm

**Pull request merged**